pkg/apis/contrail/v1alpha1/manager_types.go

```diff
@@ -23,22 +23,53 @@ type ManagerSpec struct {
-	Controls         []*Control        `json:"controls,omitempty"`
-	Kubemanagers     []*Kubemanager    `json:"kubemanagers,omitempty"`
-	Webui            *Webui            `json:"webui,omitempty"`
-	Vrouters         []*Vrouter        `json:"vrouters,omitempty"`
-	Cassandras       []*Cassandra      `json:"cassandras,omitempty"`
-	Zookeepers       []*Zookeeper      `json:"zookeepers,omitempty"`
-	Rabbitmq         *Rabbitmq         `json:"rabbitmq,omitempty"`
-	ProvisionManager *ProvisionManager `json:"provisionManager,omitempty"`
-	Command          *Command          `json:"command,omitempty"`
-	Postgres         *Postgres         `json:"postgres,omitempty"`
-	Keystone         *Keystone         `json:"keystone,omitempty"`
-	Swift            *Swift            `json:"swift,omitempty"`
-	Memcached        *Memcached        `json:"memcached,omitempty"`
-	Contrailmonitor  *Contrailmonitor  `json:"contrailmonitor,omitempty"`
-	ContrailCNIs     []*ContrailCNI    `json:"contrailCNIs,omitempty"`
+	Config           *Config               `json:"config,omitempty"`
+	Controls         []*Control            `json:"controls,omitempty"`
+	Kubemanagers     []*KubemanagerService `json:"kubemanagers,omitempty"`
+	Webui            *Webui                `json:"webui,omitempty"`
+	Vrouters         []*VrouterService     `json:"vrouters,omitempty"`
+	Cassandras       []*Cassandra          `json:"cassandras,omitempty"`
+	Zookeepers       []*Zookeeper          `json:"zookeepers,omitempty"`
+	Rabbitmq         *Rabbitmq             `json:"rabbitmq,omitempty"`
+	ProvisionManager *ProvisionManager     `json:"provisionManager,omitempty"`
+	Command          *Command              `json:"command,omitempty"`
+	Postgres         *Postgres             `json:"postgres,omitempty"`
+	Keystone         *Keystone             `json:"keystone,omitempty"`
+	Swift            *Swift                `json:"swift,omitempty"`
+	Memcached        *Memcached            `json:"memcached,omitempty"`
+	Contrailmonitor  *Contrailmonitor      `json:"contrailmonitor,omitempty"`
+	ContrailCNIs     []*ContrailCNI        `json:"contrailCNIs,omitempty"`
+}
+
+// VrouterService defines desired confgiuration of vRouter
+// +k8s:openapi-gen=true
+type VrouterService struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              VrouterServiceSpec `json:"spec,omitempty"`
+	ControlInstance   string             `json:"controlInstance,omitempty"`
+}
+
+// VrouterServiceSpec defines desired spec confgiuration of vRouter
+// +k8s:openapi-gen=true
+type VrouterServiceSpec struct {
+	CommonConfiguration  PodConfiguration     `json:"commonConfiguration,omitempty"`
+	ServiceConfiguration VrouterConfiguration `json:"serviceConfiguration"`
+}
+
+// KubemanagerService defines desired confgiuration of vRouter
+// +k8s:openapi-gen=true
+type KubemanagerService struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              KubemanagerServiceSpec `json:"spec,omitempty"`
```

Comment:
same concern like Pawel, that we should not break manager, regarding other CR the question is open if we should be able to break or not but for sure you should manipulate somehow these structures not to break the manager structure. Every breaking change should introduce a new version of API

Reply:
done